<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="freq" default="200" />
	<arg name="some_grf" 	default="some_grf"/>
	<node pkg="republisher" type="republisher_node" name="insole_republisher_$(arg some_grf)" output="screen">
		<param name="rate" 					value="$(arg freq)"/>
		<param name="input_queue_length" 			value="100"/>
		<param name="maximum_queue_length" 			value="1000"/>
		<param name="rate" 					value="$(arg freq)"/>
		<remap from="/insole_republisher_$(arg some_grf)/input" 		to="/$(arg some_grf)/insole"/>
		<remap from="/insole_republisher_$(arg some_grf)/output" 		to="/$(arg some_grf)/insole_fixed_freq"/>
		<remap from="/insole_republisher_$(arg some_grf)/wrench" 		to="/grf_$(arg some_grf)/wrench_fixed_freq"/>
		<remap from="/insole_republisher_$(arg some_grf)/wrench_filtered" to="/grf_$(arg some_grf)/wrench"/>
		<!-- Filter values for filtered output -->
		<param name="filter_output" 		value="true" />
		<param name="memory" 			value="35" />
		<param name="cutoff_freq" 		value="6" />
		<param name="delay" 			value="14" />
		<param name="spline_order" 		value="3" />
	</node>

</launch>


